### PR TITLE
[Remy9926] iP

### PR DESCRIPTION
**Pull request by Remy9926**

### CODE REVIEW

File: src/main/java/Task.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+		}
+		System.out.println("       " + this);
+		System.out.println("    ____________________________________________________________");
+	}
+	
+	public String toString() {
+		if (this.isDone()) {
+			return "[X" + "] " + description;
+		}
+		return "[ " + "] " + description;
```
Comment (bnjm2000) [inline src/main/java/Task.java]: "[X" + "] " can be combined.

File: src/main/java/Duke.java [hunk 2]
```diff
@@ -1,10 +1,121 @@
...
+    	 System.out.println("\n"
+    	 		+ "    __________                      \n"
+    	 		+ "    \\______   \\ ____   _____ ___.__.\n"
+    	 		+ "     |       _// __ \\ /     <   |  |\n"
+    	 		+ "     |    |   \\  ___/|  Y Y  \\___  |\n"
+    	 		+ "     |____|_  /\\___  >__|_|  / ____|\n"
+    	 		+ "            \\/     \\/      \\/\\/     \n"
+    	 		+ "");
+         System.out.println("    Hello! I'm Remy");
+         System.out.println("    What can I do for you?");
```
Comment (bnjm2000) [inline src/main/java/Duke.java]: great customisations

File: src/main/java/Duke.java [hunk 3]
```diff
@@ -1,10 +1,121 @@
...
+        Scanner input = new Scanner(System.in);
+        String command = input.nextLine();
+        
+        while (!command.equals("bye")) {
+        	if (command.equals("list")) {
+        		printToDoList();
+        	} else {
+        		String[] mark = command.split(" ");
+        		if (mark[0].contains("mark")) {
+            		int index = Integer.parseInt(mark[1]) - 1;
```
Comment (bnjm2000) [inline src/main/java/Duke.java]: should be indented

File: src/main/java/Duke.java [hunk 4]
```diff
@@ -1,10 +1,121 @@
...
+        			if (mark[0].equals("mark")) {
+            			toDoList[index].setDone(true);
+            		} else {
+            			toDoList[index].setDone(false);
+            		}
```
Comment (bnjm2000) [inline src/main/java/Duke.java]: should be indented

File: src/main/java/Duke.java [hunk 5]
```diff
@@ -1,10 +1,121 @@
...
+         System.out.println("    What can I do for you?");
+         printLines();
+    }
+    
+    public static void printLines() {
+    	System.out.println("    ____________________________________________________________");
+    }
+    
+    public static void addTask(Task task) {
+    	toDoList[size++] = task;
```
Comment (bnjm2000) [inline src/main/java/Duke.java]: great optimisation

File: src/main/java/Duke.java [hunk 6]
```diff
@@ -1,10 +1,121 @@
...
+    	printLines();
+    	System.out.println("     Got it. I've added this task:\n       " + task.toString() + "\n     Now you have " + size + " tasks in the list.");
```
Comment (bnjm2000) [inline src/main/java/Duke.java]: consider printing in multiple lines for better code readability

File: src/main/java/Duke.java [hunk 7]
```diff
@@ -1,10 +1,121 @@
...
+    	greetUser();
+        Scanner input = new Scanner(System.in);
+        String command = input.nextLine();
+        
+        while (!command.equals("bye")) {
+        	if (command.equals("list")) {
+        		printToDoList();
+        	} else {
+        		String[] mark = command.split(" ");
+        		if (mark[0].contains("mark")) {
```
Comment (junhyeong0411) [inline src/main/java/Duke.java]: Need to remove indent (4 spaces)

File: src/main/java/Duke.java (hunk 4 above)
Comment (junhyeong0411) [inline src/main/java/Duke.java]: Should be indented

File: src/main/java/Duke.java [hunk 8]
```diff
@@ -1,10 +1,121 @@
...
+    	int fromIndex = 0;
+    	int toIndex = 0;
+    	for (int i = 0; i < description.length; i++) {
+    		if (description[i].equals("/from")) {
+    			fromIndex = i;
+    		} else if (description[i].equals("/to")) {
+    			toIndex = i;
+    			break;
+    		}
+    	}
```
Comment (junhyeong0411) [inline src/main/java/Duke.java:202]: Good indentation

File: src/main/java/Duke.java [hunk 9]
```diff
@@ -1,10 +1,121 @@
...
+    public static void printToDoList() {
+    	printLines();
+    	System.out.println("     " + "Here's your tasks:");
+    	for (int i = 0; i < size; i++) {
+    		System.out.println("     " + Integer.toString(i + 1) + "." + toDoList[i]);
+    	}
+    	printLines();
+    }
+    
+    public static String[] getDeadlineDescription(String[] description) {
```
Comment (junhyeong0411) [inline src/main/java/Duke.java:166]: good camelCase

File: src/main/java/Task.java [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+	
+	public void setDescription(String newDescription) {
+		this.description = newDescription;
+	}
+	
+	public boolean isDone() {
+		return done;
+	}
+	
+	public void setDone(boolean status) {
```
Comment (junhyeong0411) [inline src/main/java/Task.java]: boolean variables should be named to sounds like booleans (is~~, has~~...)

File: src/main/java/Duke.java [hunk 11]
```diff
@@ -1,10 +1,121 @@
+import java.util.Arrays;
+import java.util.Scanner;
+
 public class Duke {
+	static Task[] toDoList = new Task[100];
+	static int size = 0;
```
Comment (Brian030601) [inline src/main/java/Duke.java]: Size for what? Maybe rename it as todoListSize

File: src/main/java/Duke.java (hunk 7 above)
Comment (Brian030601) [inline src/main/java/Duke.java]: Deep nesting issue here. Try narrowing the code down to one if conditionals

File: src/main/java/Duke.java [hunk 12]
```diff
@@ -1,10 +1,121 @@
...
+    public static void greetUser() {
+    	 printLines();
+    	 System.out.println("\n"
+    	 		+ "    __________                      \n"
+    	 		+ "    \\______   \\ ____   _____ ___.__.\n"
+    	 		+ "     |       _// __ \\ /     <   |  |\n"
+    	 		+ "     |    |   \\  ___/|  Y Y  \\___  |\n"
+    	 		+ "     |____|_  /\\___  >__|_|  / ____|\n"
+    	 		+ "            \\/     \\/      \\/\\/     \n"
+    	 		+ "");
```
Comment (Brian030601) [inline src/main/java/Duke.java]: This is some amazing work.

File: src/main/java/Duke.java (hunk 6 above)
Comment (Brian030601) [inline src/main/java/Duke.java]: Too much coding in one line

File: src/main/java/Task.java (hunk 10 above)
Comment (Brian030601) [reply src/main/java/Task.java]: Agreed here. Should correct the naming

File: src/main/java/Duke.java [hunk 13]
```diff
@@ -1,10 +1,121 @@
...
+    		if (description[i].equals("/from")) {
+    			fromIndex = i;
+    		} else if (description[i].equals("/to")) {
+    			toIndex = i;
+    			break;
+    		}
+    	}
+    	String desc = String.join(" ", Arrays.copyOfRange(description, 1, fromIndex));
+    	String from = String.join(" ", Arrays.copyOfRange(description, fromIndex + 1, toIndex));
+    	String to = String.join(" ", Arrays.copyOfRange(description, toIndex + 1, description.length));
```
Comment (Brian030601) [inline src/main/java/Duke.java]: consider adding comments here and there

**Review by Brian030601 — COMMENTED** [review]
Hey, overall, your program looks great. It follows the code quality guidelines. There are some minor issues, but they can be taken care of. Great work.

File: src/main/java/task/Event.java [hunk 14]
```diff
@@ -0,0 +1,24 @@
...
+	
+	public String getFrom() {
+		return from;
+	}
+	
+	public String getTo() {
+		return to;
+	}
+	
+	public String toString() {
```
Comment (J-Y-Yan) [inline src/main/java/task/Event.java:21]: It looks very good and clear of using inheritance to print message. Keep going!

File: src/main/java/Duke.java [hunk 15]
```diff
@@ -1,10 +1,200 @@
...
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+		greetUser();
+        Scanner input = new Scanner(System.in);
+        String command = input.nextLine();
+        
+        while (!command.equals("bye")) {
+        	String strippedCommand = command.strip();
+    		String[] userCmd = strippedCommand.split(" ");
+        	if (command.equals("list")) {
```
Comment (bentohset) [inline src/main/java/Duke.java]: Indentation standards does not follow the [coding standard number of spaces for indentations](https://se-education.org/guides/conventions/java/basic.html#layout). This happens throughout. Please refactor accordingly (perhaps you might need to change the settings of your IDE)

File: src/main/java/Duke.java [hunk 16]
```diff
@@ -1,10 +1,200 @@
...
+            	catch (NullPointerException e) {
+            		printLines();
+            		System.out.println("     Cannot mark/unmark a non-existent task");
+            		printLines();
+            	}
+        		catch (ArrayIndexOutOfBoundsException e) {
+        			printLines();
+            		System.out.println("     Please specify a number of the task that you want to mark/unmark");
+            		printLines();
+        		}
```
Comment (J-Y-Yan) [inline src/main/java/Duke.java]: I guess this is your coding style. I like your work but maybe it will be better to follow coding standard indentation.

**Pull request closed without merging**